frontend/src/components/Switch.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import styled from 'styled-components'
+
+const SwitchButton = styled.span<{
+  width?: number | undefined
+}>`
+  width: ${(props) => (props.width ? props.width * 0.45 + 'px' : '45px')};
+  height: ${(props) => (props.width ? props.width * 0.45 + 'px' : '45px')};
+  position: absolute;
+  left: ${(props) => (props.width ? '-' + props.width * 0.015 + 'px' : '-1.5px')};
+  border-radius: 45px;
+  border: ${(props) => (props.width ? +props.width * 0.02 + 'px solid lightgray' : '2px solid gray')};
+  transition: 0.2s;
+  background: #fff;
+`
+
+const SwitchLabel = styled.label<{
+  width?: number | undefined
+}>`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  cursor: pointer;
+  width: ${(props) => (props.width ? props.width + 'px' : '100px')};
+  height: ${(props) => (props.width ? props.width * 0.5 + 'px' : '50px')};
+  background: lightgray;
+  border-radius: 100px;
+  border: ${(props) => (props.width ? +props.width * 0.03 + 'px solid lightgray' : '2px solid gray')};
```

Comment:
here too, if they are both very similar, consider abstracting the css out using styled-components' css

Reply:
removed +

Reply:
honestly dont really know whats the best way of doing this, looked through it and tried to make one to neaten up the code, but I just see many issues as I use different numbers of 0.5, 0.3, 0.6, 0.45 which Im not too sure how to implement this solution